src/services/deviceSimulationService.js

```diff
@@ -13,18 +15,56 @@ const SIMULATION_ID = Config.simulationId;
 export class DeviceSimulationService {
 
   /**
-   * Toggles simulation status
+   * Returns a list of devicemodels
    */
-  static toggleSimulation(Etag, Enabled) {
-    return HttpClient.patch(`${ENDPOINT}simulations/${SIMULATION_ID}`, { Etag, Enabled })
-      .map(toSimulationStatusModel);
+  static getDeviceModelSelectOptions() {
+    return HttpClient.get(`${ENDPOINT}devicemodels`)
+      .map(toDeviceModelSelectOptions);
   }
 
   /**
    * Get the list of running simulated devices
    */
   static getSimulatedDevices() {
     return HttpClient.get(`${ENDPOINT}simulations/${SIMULATION_ID}`)
-      .map(toSimulationStatusModel);
+      .map(toDeviceSimulationModel);
+  }
+
+  /**
+   * Updates simulated device
+   */
+  static updateSimulatedDevices(simulation) {
+    return HttpClient.put(`${ENDPOINT}simulations/${SIMULATION_ID}`, simulation)
+      .map(toDeviceSimulationModel)
+  }
+
+  /**
+   * Toggles simulation status
+   */
+  static toggleSimulation(Etag, Enabled) {
+    return HttpClient.patch(`${ENDPOINT}simulations/${SIMULATION_ID}`, { Etag, Enabled })
+      .map(toDeviceSimulationModel);
+  }
+
+  /**
+   * Gets the simulated device models, increments the given one, then updates on the server
+   */
+  static incrementSimulatedDeviceModel(deviceModelId, increment) {
+    return this.getSimulatedDevices()
```

Comment:
Is using `this` working? I don't think it should be. I thought you have to refer to the class directly to access the static methods. E.g. `DeviceSimulationService.updateSimulatedDevices`

Reply:
Yes, it is working.  Because, um, I usually make it a practice to ensure the code is functioning prior to checking it in.  Hahaha.
But yes, I'll change it.

Reply:
Definitely not as readable even if it's working - 'this' implies an instance.